### DISCUSSION TITLE
Support multi-instanace feature

### DISCUSSION THREAD

**Discussion by MeiyongYu** [dail8859/NotepadNext#643] [Ideas]
Now notepadnext only support single-instance.
And Notepadd++, SourceInsight, Visual Code all support multi-instance.
In fact of work we will only multi project or workplace to work at diff work

The modify thinking is:
1) Keep the default mode single,
   when user choose multi-instance config than change to multi-instance and ohter instance can be to start
2) NotepadNextApplication not inherit form SingleApplication, but use composition mode.

**Comment by dail8859** [comment]
> Keep the default mode single

I think some thought needs put in to how the user specifies that a new instance should be started. Notepad++ has the option to right click the opened file and "Move to New Instance". Maybe something like that or maybe some of the other editors have good ways to let the user specify that...maybe drag and drop a file outside of the main window.

> NotepadNextApplication not inherit form SingleApplication, but use composition mode.

Why would this need changed? I think "multi-instance" needs to be thought more of "multi-window". Even with multiple windows there should only ever be one [NotepadNextApplication](https://github.com/dail8859/NotepadNext/blob/master/src/NotepadNext/NotepadNextApplication.h) instance. The application could have multiple `MainWindow`s here instead of one:

https://github.com/dail8859/NotepadNext/blob/4c7934479b01378d81acd4652e79846c685ebdbc/src/NotepadNext/NotepadNextApplication.h#L90C6-L90C36

In the past this use to be a `QList<MainWindow>` to support multiple windows, but it was simplified since multi-instance mode was able to be supported at the time.

**Reply by MeiyongYu** [reply]
Ok, I will keep the default single-instance,  and support mutli-window.
